src/Validator/OrderMailValidator.php

```diff
@@ -51,4 +51,17 @@ public function validateOrderConfMailSend($orderConfSendStatus, $orderState)
                 return false;
         }
     }
+
+    public function validateNewOrderMailSend($newOrderSendStatus, $orderState)
+    {
+        switch ($newOrderSendStatus) {
+            case \Mollie\Config\Config::NEW_ORDER_MAIL_SEND_ON_CREATION:
+                return true;
+            case \Mollie\Config\Config::NEW_ORDER_MAIL_SEND_ON_PAID:
+                return (int)Configuration::get(\Mollie\Config\Config::MOLLIE_STATUS_PAID) === (int)$orderState ||
+                    (int)Configuration::get(\Mollie\Config\Config::STATUS_PS_OS_OUTOFSTOCK_PAID) === (int)$orderState;
+            case \Mollie\Config\Config::NEW_ORDER_MAIL_SEND_ON_NEVER:
+                return false;
+        }
+    }
```

Comment:
two exactly same looking functions which accepts same arguments and they are laying in same class :thinking: 

this looks like a job for insterface to do, the code would look cleaner and more splitted then doing just like that